Compilation error after refactoring tests

The Jenkins `openmm-dev` build is failing with compilation errors (presumably introduced by #1142)
```
[ 38%] Building CXX object platforms/cuda/tests/CMakeFiles/TestCudaAndersenThermostat.dir/TestCudaAndersenThermostat.cpp.o
/home/vagrant/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaAndersenThermostat.cpp:33:10: fatal error: 'TestAndersenThermostat.h' file not found
#include "TestAndersenThermostat.h"
         ^
1 error generated.
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaAndersenThermostat.dir/TestCudaAndersenThermostat.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaAndersenThermostat.dir/all] Error 2
make: *** [all] Error 2
```
See [full build logs](https://jenkins.choderalab.org/job/conda-openmm-dev-linux-vagrant-2/125/console).

I can confirm this, I cannot build the latest master:

```
~/src/openmm/platforms/cuda/tests/TestCudaCMAPTorsionForce.cpp:33:34: fatal error: TestCMAPTorsionForce.h: No such file or directory
 #include "TestCMAPTorsionForce.h"
                                  ^
compilation terminated.
~/src/openmm/openmm/platforms/cuda/tests/TestCudaAndersenThermostat.cpp:33:36: fatal error: TestAndersenThermostat.h: No such file or directory
 #include "TestAndersenThermostat.h"
                                    ^
compilation terminated.
~/src/openmm/openmm/platforms/cuda/tests/TestCudaCMMotionRemover.cpp:33:33: fatal error: TestCMMotionRemover.h: No such file or directory
 #include "TestCMMotionRemover.h"
                                 ^
compilation terminated.
~/src/openmm/openmm/platforms/cuda/tests/TestCudaBrownianIntegrator.cpp:33:36: fatal error: TestBrownianIntegrator.h: No such file or directory
 #include "TestBrownianIntegrator.h"
```

It fails on `osx` too:
```
Linking CXX shared library ../../../libOpenMMOpenCL.dylib
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaCMMotionRemover.cpp:33:10: fatal error: 'TestCMMotionRemover.h' file not found
#include "TestCMMotionRemover.h"
         ^
1 error generated.
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCMMotionRemover.dir/TestCudaCMMotionRemover.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCMMotionRemover.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaCustomBondForce.cpp:33:10: fatal error: 'TestCustomBondForce.h' file not found
#include "TestCustomBondForce.h"
         ^
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaCustomAngleForce.cpp:33:10: fatal error: 'TestCustomAngleForce.h' file not found
#include "TestCustomAngleForce.h"
         ^
1 error generated.
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCustomBondForce.dir/TestCudaCustomBondForce.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCustomBondForce.dir/all] Error 2
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaCheckpoints.cpp:33:10: fatal error: 'TestCheckpoints.h' file not found
#include "TestCheckpoints.h"
         ^
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaAndersenThermostat.cpp:33:10: fatal error: 'TestAndersenThermostat.h' file not found
#include "TestAndersenThermostat.h"
         ^
1 error generated.
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaCMAPTorsionForce.cpp:33:10: fatal error: 'TestCMAPTorsionForce.h' file not found
#include "TestCMAPTorsionForce.h"
         ^
/Users/jenkins-slave/workspace/conda-openmm-dev-osx-2/miniconda/conda-bld/work/platforms/cuda/tests/TestCudaBrownianIntegrator.cpp:33:10: fatal error: 'TestBrownianIntegrator.h' file not found
#include "TestBrownianIntegrator.h"
         ^
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCustomAngleForce.dir/TestCudaCustomAngleForce.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCustomAngleForce.dir/all] Error 2
[ 86%] 1 error generated.
Built target OpenMMOpenCL
1 error generated.
1 error generated.
1 error generated.
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaAndersenThermostat.dir/TestCudaAndersenThermostat.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaAndersenThermostat.dir/all] Error 2
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCheckpoints.dir/TestCudaCheckpoints.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCheckpoints.dir/all] Error 2
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCMAPTorsionForce.dir/TestCudaCMAPTorsionForce.cpp.o] Error 1
make[2]: *** [platforms/cuda/tests/CMakeFiles/TestCudaBrownianIntegrator.dir/TestCudaBrownianIntegrator.cpp.o] Error 1
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaCMAPTorsionForce.dir/all] Error 2
make[1]: *** [platforms/cuda/tests/CMakeFiles/TestCudaBrownianIntegrator.dir/all] Error 2
make: *** [all] Error 2
```
Full build logs available [here](https://jenkins.choderalab.org/job/conda-openmm-dev-osx-2/130/console).

--- a/platforms/cuda/CMakeLists.txt
+++ b/platforms/cuda/CMakeLists.txt
@@ -13,9 +13,9 @@
 #----------------------------------------------------
 
 set(OPENMM_BUILD_CUDA_TESTS TRUE CACHE BOOL "Whether to build CUDA test cases")
-if(OPENMM_BUILD_CUDA_TESTS)
+if(BUILD_TESTING AND OPENMM_BUILD_CUDA_TESTS)
     SUBDIRS (tests)
-endif(OPENMM_BUILD_CUDA_TESTS)
+endif(BUILD_TESTING AND OPENMM_BUILD_CUDA_TESTS)
 
 # The source is organized into subdirectories, but we handle them all from
 # this CMakeLists file rather than letting CMake visit them as SUBDIRS.

--- a/platforms/opencl/CMakeLists.txt
+++ b/platforms/opencl/CMakeLists.txt
@@ -13,9 +13,9 @@
 #----------------------------------------------------
 
 set(OPENMM_BUILD_OPENCL_TESTS TRUE CACHE BOOL "Whether to build OpenCL test cases")
-if(OPENMM_BUILD_OPENCL_TESTS)
+if(BUILD_TESTING AND OPENMM_BUILD_OPENCL_TESTS)
     SUBDIRS (tests)
-endif(OPENMM_BUILD_OPENCL_TESTS)
+endif(BUILD_TESTING AND OPENMM_BUILD_OPENCL_TESTS)
 
 # The source is organized into subdirectories, but we handle them all from
 # this CMakeLists file rather than letting CMake visit them as SUBDIRS.

